src/Xablu.WebApiClient/BaseClient.cs

```diff
@@ -19,29 +18,7 @@ public BaseClient(IWebApiClient apiClient)
 
         protected async Task<TResult> ExecuteRemoteRequest<TResult>(Func<Task<TResult>> action)
         {
-            TResult result = default(TResult);
-
-            try
-            {
-                result = await Policy
-                    .Handle<WebException>()
-                    .WaitAndRetryAsync
-                    (
-                        retryCount: 5,
-                        sleepDurationProvider: retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt))
-                    )
-                    .ExecuteAsync(action).ConfigureAwait(false);
-            }
-            catch (Exception e)
-            {
-                if (e.GetType().Namespace == "Java.Net")
-                {
-                    throw new HttpExtensions.HttpResponseException(HttpStatusCode.RequestTimeout, e.Message, e.StackTrace);
-                }
-                throw;
-            }
-
-            return result;
+            return await WebApiClientPollyExtensions.PollyDecorator(action, 5, 5);
```

Comment:
I also think 5 is a bit long. In the previous version this was actually increased in an interval (`TimeSpan.FromSeconds(Math.Pow(2, retryAttempt))`). But this implies you'll need to supply a function to the method which I left out to keep it simple. Maybe I can add another overload so you could pick between simply waiting for some seconds or you can supply a function to make a more complicated calculation.

Regarding the other comment, I think you are right and will simply return the task.